### PR TITLE
Lambda search API documentation

### PR DESCRIPTION
**Pull request by mmanes**
Add doc for `/api/lambda/search`

## Related PRs
- https://github.com/FusionAuth/fusionauth-app/pull/203
- https://github.com/FusionAuth/fusionauth-client-builder/pull/62

### CODE REVIEW

**Review by mooreds — CHANGES_REQUESTED** [review]
response body shows up 2x when rendered on the website, didn't dig into why.

A few other suggestions.

File: site/docs/v1/tech/apis/_lambda-search-request-parameters.adoc [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * `engineType` - the JavaScript execution engine for the Lambda
++
+The order direction is optional. Possible values of the order direction are `ASC` or `DESC`. If omitted, the default sort order is `ASC`.
++
+For example, to order the results by the insert instant in a descending order, use `insertInstant DESC`.
+
+[field]#{parameter_prefix}startRow# [type]#[Integer]# [optional]#Optional# [default]#defaults to `0`#::
+The offset into the total results. In order to paginate the results, increment this value by the [field]#numberOfResults# for subsequent requests.
++
+For example, if the total search results are greater than the page size designated by [field]#numberOfResults#, set this value to `25` to retrieve results `26-50`, assuming the default page size.
```
[the remark this answers is not in the document]
Comment (spwitt) [reply site/docs/v1/tech/apis/_lambda-search-request-parameters.adoc:32]: I've rolled across all my PRs. I believe that Mark has done the same.

[the remark this answers is not in the document]
Comment (mmanes) [reply site/docs/v1/tech/apis/_lambda-search-request-parameters.adoc:32]: Yep, that should be the same across all of mine now.

File: site/docs/v1/tech/apis/_lambda-responses-body.adoc [hunk 2]
```diff
@@ -4,32 +4,32 @@
...
+[field]#lambdas``[x]``.body# [type]#[String]#::
 The lambda function body, a JavaScript function.
 
-[field]#lambda``[x]``.debug# [type]#[Boolean]#::
+[field]#lambdas``[x]``.debug# [type]#[Boolean]#::
 Whether or not debug event logging is enabled for this Lambda.
 
-[field]#lambda``[x]``.enabled# [type]#[Boolean]# [deprecated]#Deprecated#::
+[field]#lambdas``[x]``.enabled# [type]#[Boolean]# [deprecated]#Deprecated#::
 Whether or not this Lambda is enabled.
```
[the remark this answers is not in the document]
Comment (mmanes) [reply site/docs/v1/tech/apis/_lambda-responses-body.adoc:14]: I'm happy to pull it, but I'd defer that decision to someone other than me 😃

File: site/docs/src/json/lambdas/search-response.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "lambdas": [
+    {
+      "id": "7e66bac3-fa41-47fb-b8fd-12b35b5e1807",
+      "body": "function reconcile(one, two) { console.log('Hello world'); }",
```
[the remark this answers is not in the document]
Comment (mmanes) [reply site/docs/src/json/lambdas/search-response.json]: For consistency, I've updated all of the requests and responses to include this more realistic function body.

File: site/docs/v1/tech/apis/_lambda-responses-body.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (mooreds) [reply site/docs/v1/tech/apis/_lambda-responses-body.adoc:14]: Let's not, then. Sometime though, probably as a sep PR.

**Pull request merged**